Source/AudioPlayer.swift

```diff
@@ -125,52 +58,68 @@ public class AudioPlayer {
             }
         }
         
-        guard let startTime = players.first!.lastRenderTime?.sampleTime else { return }
-        startTimer()
-        players.forEach { $0.play(at: AVAudioTime(sampleTime: startTime, atRate: Double(audioSampleRate))) }
+        timeline.start()
+        players.forEach {
+            guard let playbackTime = $0.playbackTime else { return }
+            $0.play(at: playbackTime)
+        }
     }
     
     public func pause() {
         guard isPlaying else { return }
-        cancelTimer()
+        timeline.pause()
         players.forEach { $0.pause() }
     }
     
     public func stop() {
+        timeline.reset()
         players.forEach { $0.stop() }
     }
     
-    public func scheduleFiles() {
-        schedulers.forEach(schedule)
+    public func reload() {
+        timeline.reset()
+        players.forEach { $0.reload() }
+    }
+    
+    public func seek(to second: Float) {
+        timeline.seek(Double(second))
+        let currrentSecond = max(min(Float(timeline.currentTime), audioLengthSeconds), 0)
```

Comment:
isn't there a better (more expressive) way to do this? if not, please add a comment here specifying the intent

Reply:
Cretate extension function of Float `bound(by: Range) -> Float`
`timeline.currentTime.bound(by: [0...length])`